package Protocol.Data.GroupChat;

message Packet {
    optional GroupMessage group_message = 1;
    optional GroupMessageAcknowledge group_message_acknowledge = 2;
}

message GroupMessage {
    // signature = message_id + message_text + timestamp.toString
    optional uint32 message_id = 1;
    optional bytes signature = 2;
    optional int64 timestamp = 3;
    optional string author = 4; // ricochet id like "ricochet:xxxxxxxxxxxxxxxx"
    optional string message_text = 5;
}

message GroupMessageAcknowledge {
    optional uint32 message_id = 1;
    optional bool accepted = 2 [default = true];
}

